### PR TITLE
[Storage][Blob][Bug]Support parsing blob url with '/' in blob name

### PR DESCRIPTION
**Pull request by xiafu-msft**
https://github.com/Azure/azure-sdk-for-python/issues/12563
https://github.com/Azure/azure-sdk-for-python/issues/12568

### CODE REVIEW

File: sdk/storage/azure-storage-queue/azure/storage/queue/_shared/base_client.py [hunk 1]
```diff
@@ -320,11 +325,11 @@ def __exit__(self, *args):  # pylint: disable=arguments-differ
         pass
 
 
-def format_shared_key_credential(account, credential):
+def format_shared_key_credential(account_name, credential):
```
Comment (rakshith91) [inline sdk/storage/azure-storage-queue/azure/storage/queue/_shared/base_client.py:328]: we should rename this to `_format_shared_key_credential` . It's not exposed so rather make it private

Comment (xiafu-msft) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_shared/base_client.py]: done!

File: sdk/storage/azure-storage-blob/tests/test_blob_client.py [hunk 2]
```diff
@@ -438,6 +439,31 @@ def test_create_service_with_custom_account_endpoint_path(self):
...
+        blob_emulator_url = 'http://127.0.0.1:1000/devstoreaccount1/containername/dir1/sub000/2010_Unit150_Ivan097_img0003.jpg'
+        blob_client = BlobClient.from_blob_url(blob_emulator_url)
+        self.assertEqual(blob_client.container_name, "containername")
+        self.assertEqual(blob_client.blob_name, "dir1/sub000/2010_Unit150_Ivan097_img0003.jpg")
+
+    def test_create_client_for_emulator(self):
+        container_client = ContainerClient(
+            account_url='http://127.0.0.1:1000/devstoreaccount1',
+            container_name='newcontainer',
+            credential='Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==')
```
Comment (rakshith91) [inline sdk/storage/azure-storage-blob/tests/test_blob_client.py:457]: hoping this is not real :P

Comment (xiafu-msft) [reply sdk/storage/azure-storage-blob/tests/test_blob_client.py:457]: it's not

Comment (kasobol-msft) [reply sdk/storage/azure-storage-blob/tests/test_blob_client.py:457]: I bet we'll get email to remove it soon :P

**Pull request merged**